
The development of long-read RNA-seq technologies has enabled the profiling of full-length reads while mitigating biases in previous generations of RNA-seq technologies, improving the accuracy of isoform abundance estimates. However, biases present in long-read RNA-seq data and their effects on transcript quantification have not yet been extensively explored in the literature. 

In this thesis, we examine \textit{degradation bias} present in long-read direct RNA-seq, where reads are truncated and map to multiple isoforms, leading to ambiguity in read-to-transcript assignment and erroneous isoform abundance estimates. We characterise degradation in real datasets, develop a bias-aware model for transcript quantification and develop expectation maximization and variational Bayesian algorithms for inferring isoform abundance estimates. By accounting for degradation bias, we demonstrate improvements in transcript quantification on simulated datasets with known degradation rate and real datasets with sequencing spike-ins.

