In this project, we consider the problem of transcript isoform quantification from long-read RNA-sequencing. We assume that raw signal from the sequencing platform has been decoded through basecalling, and that reads have been aligned to a reference. Our focus is on the assignment of aligned reads to transcript isoforms. In particular, we examine a form of bias present in long-read RNA-seq referred to as \textit{degradation bias}, and characterise its effects on the assignment of reads to isoforms. 

We propose methods to learn the degradation bias from RNA-seq data, and demonstrate that such bias is sample- and protocol-specific and if not corrected for, leads to erroneous quantification estimates. We suggest a mixture model on the observed read length distribution to correct for this degradation bias, and incorporate this model as part of \texttt{bambu}, a tool for context-aware transcript discovery and quantification from long-read RNA-seq. By accounting for degradation bias, we demonstrate improvements in quantification on simulated RNA-seq datasets. 

In addition, we describe a variational Bayesian algorithm for quantification that extends the current expectation-maximisation algorithm implemented in \texttt{bambu}. The variational Bayesian algorithm has faster convergence and identifies full posterior distributions over quantification estimates, providing a measure of uncertainty of each estimate through credible intervals.  

