
Third generation sequencing technologies have enabled the production of long reads ranging from tens to hundreds of kilobases in length \cite{Pollard2018}, and have shown promise in resolving many challenges in genomics and transcriptomics \cite{Bolisetty2015, Byrne2017, DeCoster2019, Liu2019, Mantere2019, Nurk2021}. In particular, long-read technologies enable greater insight into the transcriptome and its complexity, which is crucial in understanding the functioning of cells and their biological processes. These technologies allow accurate detection of a larger proportion of full-length transcripts and novel splice junctions while mitigating biases associated with short-read technologies, enabling more accurate abundance estimates of reference and novel isoforms. Nevertheless, biases are still present in long-read technologies, albeit to a lesser extent. 

This thesis focuses on a particular bias present in long-read direct RNA-seq referred to as \textit{degradation bias}. This bias arises due to the fact that from the time a transcript is generated to when it is sequenced, it is subject to multiple factors that results in its degradation. Consequently, the reads obtained from sequencing are often truncated, resulting in ambiguity in read assignment to transcript isoforms. This, in turn, leads to erroneous isoform abundance estimates. 

In this thesis, we attempt to characterise degradation bias and its effects on quantification from long-read direct RNA-seq, and develop a framework to model and correct such bias for accurate transcript quantification and isoform abundance estimation.

\section{Review}

Here, we review various concepts and existing literature relevant to our aim of modeling bias in long-read RNA-seq. We first examine (i) biases in short-read RNA-seq technologies and how they are accounted for by existing methods, which provide useful ideas on how to handle biases in long-read RNA-seq. Next, we review (ii) long-read RNA-seq technologies and how they mitigate biases in (i), and (iii) biases that long-read technologies themselves possess. 

\subsection{Bias modeling in short-read RNA-seq}

Short-read RNA-seq technologies enable deep sequencing of highly accurate short reads, and has been the dominant technology for profiling the transcriptome since its popularisation in the early 2010s \cite{Lowe2017}. Typically, library preparation protocol for an RNA-seq experiment following \gls{rna} extraction involves \gls{rna} fragmentation, followed by the use of random hexamer primers for priming of the fragments to synthesize one strand of \gls{cdna}. After second strand synthesis, the resulting double stranded \gls{cdna} are size-selected and amplified via \gls{pcr} amplification to generate enough \gls{cdna} for sequencing \cite{Marguerat2010}.  

Biases in short-read RNA-seq have been extensively studied \cite{Hansen2010, Li2010, Li2011, Zhengpeng2010, Roberts2011, Benjamini2012, Lahens2014, Love2016}, and can often be traced back to specific steps in the protocol described above. For instance, \cite{Hansen2010} showed that priming with random hexamer primers induces bias in the nucleotide composition of the reads and results in non-uniform representation of reads across the length of the transcript. In addition, size selection of fragments results in an over-representation of fragments of a certain length, while \gls{rna} degradation and \gls{mrna} selection can lead to over-representation of fragments that are located towards either the beginning or end of the transcript \cite{Roberts2011, Lahens2014, Love2016}. \gls{pcr} amplification before sequencing also introduces bias by preferential amplification of fragments with certain GC content \cite{Benjamini2012, Love2016}. The combination of these biases affect quantification estimates, and if not corrected for, leads to erroneous estimates \cite{Roberts2011, Love2016}. 

Existing methods to address these biases can be categorised into two broad classes. The first class of methods involve innovations in library preparation methods to reduce bias from their source of origin \cite{Vandijk2014}. For instance, an amplification-free RNA-seq protocol was proposed in \cite{Mamanova2010} to reduce amplification bias, while thermostable polymerases that exhibit relatively lower GC bias were identified in \cite{Quail2012}. We focus on the second class of methods that involve modeling and correcting for the bias \textit{in silico}. In particular, here we focus on the approaches in the literature that are most relevant to our aim. 

\subsection{Long-read technologies}

As the name suggests, long-read sequencing technologies allow the sequencing of long reads ranging from one to hundreds of kilobases, improving over short-read technologies that yield fragments ranging from 50 to 600 bp. This has been made possible by the development of biophysical techniques to capture full-length DNA and RNA \cite{Weirather2017}. While the increased read length provided by these technologies has enabled leaps in genomics and transcriptomics research \cite{Bolisetty2015, Byrne2017, DeCoster2019, Liu2019, Mantere2019, Nurk2021}, these same technologies suffer from high sequencing error rates and low throughput \cite{Carneiro2012, Reuter2015, Jain2015}.  

Two of the most widely used platforms for long-read sequencing include Pacific Biosciences (PacBio) and Oxford Nanopore Technologies (ONT). We omit discussion on PacBio data for the time being as we work only with ONT data in this thesis. ONT is a nanopore-based sequencing technology that currently provides three protocols for transcriptomic sequencing: direct RNA, direct cDNA and PCR-cDNA sequencing. In all three protocols, DNA or RNA is processed and ligated with sequencing adaptors comprising a motor protein that ensures unidirectional single-nucleotide translocation through the nanopore on the order of milliseconds \cite{Jain2016}. Each nanopore is associated with a sensor that measures changes in ionic current caused by differences in the nucleotides occupying the pore \cite{Jain2016}, giving rise to raw electric current signal. In addition, the nanopores are embedded in a synthetic electrically-resistant membrane to ensure that all current only passes through the pore, resulting in a cleaner signal \cite{Wang2021}. The raw electric current signal is then translated to nucleotide sequence via base-calling algorithms \cite{Rang2018}. 

We briefly describe the similarities and differences between the three ONT protocols, with a focus on direct RNA-seq. In direct RNA-seq (Fig. \ref{fig:sec-1-drna}), native poly(A) RNA molecules are enriched and directly sequenced from the 3' end. An optional reverse transcription step stabilizes the RNA strand by reducing intramoleulcar secondary structure of the RNA, resulting in better sequencing output \cite{Garalde2018}. In direct cDNA and PCR-cDNA protocols, poly(A) RNA is reverse transcribed followed by second strand synthesis to produce double stranded cDNA. Strand-switching is used to increase the proportion of full-length cDNAs. In PCR-cDNA, double stranded cDNAs are amplified prior to sequencing, giving higher throughput. Across all three protocols, GC bias was evaluated to be minimal in the data, including data from the PCR-cDNA protocol, provided that the number of PCR cycles was capped \cite{ONT}. 

\begin{figure}[H]
    \centering
    \includegraphics[width=\textwidth]{figures/sec-1-drna.png}
    \caption[ONT direct RNA-sequencing protocol]{ONT direct RNA-sequencing protocol. \textbf{a.} Native RNA with 3' poly(A) tails are enriched for sequencing. poly(A) tails can be added to non-poly(A) RNA with a poly(A)-tailing kit. \textbf{b.} A poly(T) adaptor is annealed to native RNA to prime first-strand synthesis of cDNA. \textbf{c.} Reverse transcription of the RNA strand stabilises it for sequencing and improves sequencing output. \textbf{d.} A sequencing adaptor comprising a motor protein is ligated to the RNA strand and is necessary for guiding it through the nanopore. \textbf{e.} RNA traverses through the nanopore, producing raw signal output which is then basecalled.}
    \label{fig:sec-1-drna}
\end{figure}

Compared to cDNA protocols, direct RNA-seq mitigates any form of bias or errors due to PCR or reverse transcription \cite{Garalde2018, Depledge2019} which is important for obtaining accurate and reliable isoform abundance estimates. Despite this, a recent evaluation of ONT direct RNA-seq showed that although direct RNA-seq yielded significantly longer reads compared to short-read RNA-seq, the proportion of reads corresponding to full-length transcripts remained low due to read truncation \cite{Soneson2019}. Assigning direct RNA-seq reads to their isoform of origins remains highly nontrivial, in large part due to the number of secondary alignments for each read \cite{Soneson2019}. In the following section, we review some possible explanations for the observed read truncation.   

\subsection{Biases in long-read RNA-seq}

Factors that influence transcript degradation can be broadly classified as being intra-cellular or extra-cellular. 

\paragraph{Intra-cellular degradation} Degradation of transcripts can occur within the cell due to the action of both exo- and endoribonucleases. Exoribonuclease cleavage from the 5’ or 3’ ends of the transcript. Co-transcriptional processes such as 5’-capping or 3’-polyadenylation help to stabilise the transcript for translation initiation. The removal of these modifications can initiate transcript degradation. Exoribonuclease-mediated degradation can occur in both the 5’-3’ or 3’-5’ directions. Endoribonuclease cleavage within the body of the transcript (this occurs to a lesser extent). Cleavage within the transcript body exposes unprotected 5’ and 3’ ends of the distal and proximal segments of the cleaved transcript, which then expose these segments to degradation by exoribonucleases. 

\paragraph{Extra-cellular degradation} Steps in the library preparation and sequencing can also affect the degradation of transcripts. Fragmentation of transcripts from library preparation. Truncation of transcripts from the 5’ end due to artifacts of nanopore sequencing (e.g. pore blocking). Since polyA+ transcripts are sequenced from the 3’ end, sequencing artifacts result in truncation from the 5’ end.

%https://pubmed.ncbi.nlm.nih.gov/12902380/
%https://pubmed.ncbi.nlm.nih.gov/21516085/
%https://www.nature.com/articles/nrm2104
%https://bmcbiol.biomedcentral.com/articles/10.1186/1741-7007-12-42
%https://www-nature-com.libproxy1.nus.edu.sg/articles/s41592-019-0617-2#Abs1

\section{Organisation}

The chapters of this thesis are organised as follows:
\begin{itemize}
    \item In Chapter 2, we formalise the notion of degradation, and characterise degradation in long-read direct RNA-seq data in reference isoforms and sequencing spike-ins across multiple cell lines and samples. 
    \item In Chapter 3, we develop a generative model for degradation-aware isoform abundance estimates and derive expectation maximization and variational inference algorithms for parameter inference. 
    \item In Chapter 4, we evaluate our model and inference algorithms on simulated datasets with known degradation and real datasets with sequencing spike-ins. We benchmark our model against existing long-read transcript quantification methods in the literature. 
    \item In Chapter 5, we summarise the ideas of this thesis and discuss potential directions of future work. 
\end{itemize}