
Third generation sequencing technologies have enabled the production of long reads ranging from tens to hundreds of kilobases in length \cite{Pollard2018}, and have shown promise in resolving many challenges in genomics and transcriptomics \cite{Bolisetty2015, Byrne2017, DeCoster2019, Liu2019, Mantere2019, Nurk2021}. In particular, long-read technologies enable greater insight into the transcriptome and its complexity, which is crucial in understanding the functioning of cells and their biological processes. These technologies allow accurate detection of a larger proportion of full-length transcripts and novel splice junctions while mitigating biases associated with short-read technologies, enabling more accurate abundance estimates of reference and novel isoforms. Nevertheless, biases are still present in long-read technologies, albeit to a lesser extent. 

This thesis focuses on a particular bias present in long-read direct RNA-seq referred to as \textit{degradation bias}. This bias arises due to the fact that from the time a transcript is generated to when it is sequenced, it is subject to multiple factors that results in its degradation. Consequently, the reads obtained from sequencing are often truncated, resulting in ambiguity in read assignment to transcript isoforms. This, in turn, leads to erroneous isoform abundance estimates. 

In this thesis, we attempt to characterise degradation bias and its effects on quantification from long-read direct RNA-seq, and develop a framework to model and correct such bias for accurate transcript quantification and isoform abundance estimation.

\section{Review}

Here, we review various concepts and existing literature relevant to our aim of modeling bias in long-read RNA-seq. We first examine (i) biases in short-read RNA-seq technologies and how they are accounted for by existing methods, which provide useful ideas on how to handle biases in long-read RNA-seq. Next, we review (ii) long-read RNA-seq technologies and how they mitigate biases in (i), and (iii) biases that long-read technologies themselves possess. 

\subsection{Bias modeling in short-read RNA-seq}

Short-read RNA-seq technologies enable deep sequencing of highly accurate short reads, and has been the dominant technology for profiling the transcriptome since its popularisation in the early 2010s \cite{Lowe2017}. Typically, library preparation protocol for an RNA-seq experiment following \gls{rna} extraction involves \gls{rna} fragmentation, followed by the use of random hexamer primers for priming of the fragments to synthesize one strand of \gls{cdna}. After second strand synthesis, the resulting double stranded \gls{cdna} are size-selected and amplified via \gls{pcr} amplification to generate enough \gls{cdna} for sequencing \cite{Marguerat2010}.  

Biases in short-read RNA-seq have been extensively studied \cite{Hansen2010, Li2010, Li2011, Zhengpeng2010, Roberts2011, Benjamini2012, Lahens2014, Love2016}, and can often be traced back to specific steps in the protocol described above. For instance, \cite{Hansen2010} showed that priming with random hexamer primers induces bias in the nucleotide composition of the reads and results in non-uniform representation of reads across the length of the transcript. In addition, size selection of fragments results in an over-representation of fragments of a certain length, while \gls{rna} degradation and \gls{mrna} selection can lead to over-representation of fragments that are located towards either the beginning or end of the transcript \cite{Roberts2011, Lahens2014, Love2016}. \gls{pcr} amplification before sequencing also introduces bias by preferential amplification of fragments with certain GC content \cite{Benjamini2012, Love2016}. The combination of these biases affect quantification estimates, and if not corrected for, leads to erroneous estimates \cite{Roberts2011, Love2016}. 

Existing methods to address these biases can be categorised into two broad classes. The first class of methods involve innovations in library preparation methods to reduce bias from their source of origin \cite{Vandijk2014}. For instance, an amplification-free RNA-seq protocol was proposed in \cite{Mamanova2010} to reduce amplification bias, while thermostable polymerases that exhibit relatively lower GC bias were identified in \cite{Quail2012}. We focus on the second class of methods that involve modeling and correcting for the bias \textit{in silico}. In particular, we focus on the approaches in the literature that are most relevant to our aim. 

% Single-isoform genes
In modeling bias, one needs to estimate parameters describing the bias from the data. The first approach adopted by many short-read RNA-seq methods for modeling bias is to use single isoform genes for estimating bias parameters. For instance, 
Cufflinks corrects for fragment length bias by estimating fragment length distributions based on a set of single isoform gene \cite{Roberts2011}, while alpine  estimates bias offsets and coefficients from single isoform genes for fragment length, read start sequence preference and GC content \cite{Love2016}. This approach ensures that the bias terms are computed from a subset of the data with the least ambiguity in assignment to the correct isoform. Furthermore, due to the complexity of real data, non-parametric density estimation is often used to model certain features. Again, both Cufflinks and alpine use kernel density estimates to fit an empirical fragment length distribution to correct for positional bias, i.e., the enrichment of fragments at the start or end of the transcripts. RSEM uses an empirical distribution to model the position of reads within fragments \cite{Li2011}. Other complex parametric approaches have been used, such as a mixture of Gaussian in \cite{MIXMIX}, but appear less popular possibly due to parametric constraints. Last and perhaps most relevant is the use of curves to characterise and model for bias. In \cite{NURD}, coverage over single isoform genes is used to compute global and local bias curves which are then used to describe the non-uniformity of read distributions across all genes and for each gene respectively. In particular, the coverage over a gene is binned and transformed into a step function over exons of a gene \cite{NURD}. We consider all these approaches when characterising bias and modeling for it in the following chapters. 

\subsection{Long-read technologies}

As the name suggests, long-read sequencing technologies allow the sequencing of long reads ranging from one to hundreds of kilobases, improving over short-read technologies that yield fragments ranging from 50 to 600 bp. This has been made possible by the development of biophysical techniques to capture full-length DNA and RNA \cite{Weirather2017}. While the increased read length provided by these technologies has enabled leaps in genomics and transcriptomics research \cite{Bolisetty2015, Byrne2017, DeCoster2019, Liu2019, Mantere2019, Nurk2021}, these same technologies suffer from high sequencing error rates and low throughput \cite{Carneiro2012, Reuter2015, Jain2015}.  

Two of the most widely used platforms for long-read sequencing include Pacific Biosciences (PacBio) and Oxford Nanopore Technologies (ONT). We omit discussion on PacBio data for the time being as we work only with ONT data in this thesis. ONT is a nanopore-based sequencing technology that currently provides three protocols for transcriptomic sequencing: direct RNA, direct cDNA and PCR-cDNA sequencing. In all three protocols, DNA or RNA is processed and ligated with sequencing adaptors comprising a motor protein that ensures unidirectional single-nucleotide translocation through the nanopore on the order of milliseconds \cite{Jain2016}. Each nanopore is associated with a sensor that measures changes in ionic current caused by differences in the nucleotides occupying the pore \cite{Jain2016}, giving rise to raw electric current signal. In addition, the nanopores are embedded in a synthetic electrically-resistant membrane to ensure that all current only passes through the pore, resulting in a cleaner signal \cite{Wang2021}. The raw electric current signal is then translated to nucleotide sequence via base-calling algorithms \cite{Rang2018}. 

We briefly describe the similarities and differences between the three ONT protocols, with a focus on direct RNA-seq. In direct RNA-seq (Fig. \ref{fig:sec-1-drna}), native poly(A) RNA molecules are enriched and directly sequenced from the 3' end. An optional reverse transcription step stabilizes the RNA strand by reducing intramoleulcar secondary structure of the RNA, resulting in better sequencing output \cite{Garalde2018}. In direct cDNA and PCR-cDNA protocols, poly(A) RNA is reverse transcribed followed by second strand synthesis to produce double stranded cDNA. Strand-switching is used to increase the proportion of full-length cDNAs. In PCR-cDNA, double stranded cDNAs are amplified prior to sequencing, giving higher throughput. Across all three protocols, GC bias was evaluated to be minimal in the data, including data from the PCR-cDNA protocol, provided that the number of PCR cycles was capped \cite{ONT}. 

Compared to cDNA protocols, direct RNA-seq mitigates any form of bias or errors due to PCR or reverse transcription \cite{Garalde2018, Depledge2019} which is important for obtaining accurate and reliable isoform abundance estimates. Despite this, a recent evaluation of ONT direct RNA-seq showed that although direct RNA-seq yielded significantly longer reads compared to short-read RNA-seq, the proportion of reads corresponding to full-length transcripts remained low due to read truncation \cite{Soneson2019}. Assigning direct RNA-seq reads to their isoform of origins remains highly nontrivial, in large part due to the number of secondary alignments for each read \cite{Soneson2019}. In the following section, we review some possible explanations for the observed read truncation.   

\begin{figure}[H]
    \centering
    \includegraphics[width=\textwidth]{figures/sec-1-drna.png}
    \caption[ONT direct RNA-sequencing protocol]{ONT direct RNA-sequencing protocol. \textbf{a.} Native RNA with 3' poly(A) tails are enriched for sequencing. poly(A) tails can be added to non-poly(A) RNA with a poly(A)-tailing kit. \textbf{b.} A poly(T) adaptor is annealed to native RNA to prime first-strand synthesis of cDNA. \textbf{c.} Reverse transcription of the RNA strand stabilises it for sequencing and improves sequencing output. \textbf{d.} A sequencing adaptor comprising a motor protein is ligated to the RNA strand and is necessary for guiding it through the nanopore. \textbf{e.} RNA traverses through the nanopore, producing raw signal output which is then basecalled.}
    \label{fig:sec-1-drna}
\end{figure}

\subsection{Biases in long-read RNA-seq}

We now discuss biases in long-read ONT data with a particular focus on degradation bias observed in the direct RNA protocol.

\paragraph{GC bias} According to ONT, datasets sequenced with all three protocols (direct RNA, direct cDNA, PCR cDNA) exhibit virtually no GC bias \cite{ONTGC}, with low correlations between the GC content and read count for each gene. This was corroborated by studies finding that compared to other high throughput sequencing methods such as short-read RNA-seq or PacBio sequencing, there was no relative coverage biases across regions of different GC content \cite{Nookaew2017, HTSGC}. 

\paragraph{Degradation bias} Here, we consider sources of degradation bias observed in long-read direct RNA-seq, where the reads are degraded or truncated \cite{Soneson2019, Workman2019}. This results in ambiguity in read-to-isoform assignment \cite{Soneson2019} (Fig. \ref{fig:degbias}). 

Broadly, factors that influence transcript degradation can be broadly classified as being intra-cellular (\textit{in vivo}) or extra-cellular (\textit{in vitro}). We first examine the dominant source of degradation \textit{in vivo}, which is RNA decay. RNA decay is a well studied phenomena, and can occur due to the action of both exo- and endoribonucleases \cite{CONTI2005, Garneau2007, HOUSELEY2009, Keene2010}. In the first instance, RNA decay occurs due to exoribonuclease cleavage from the 5’ or 3’ ends of the transcript. In mRNAs, this results in the removal of modifications such as 5’-capping or 3’-polyadenylation that help to stabilise the transcript for translation initiation, initiating RNA degradation. Such degradation can occur in both the 5’-3’ or 3’-5’ directions, with the former being the more dominant pathway \cite{Garneau2007}.  

\begin{figure}[H]
    \centering
    \includegraphics[width=\textwidth]{figures/sec-1-degbias.png}
    \caption[Degradation bias results in multi-mapping reads]{Degradation bias results in multi-mapping reads. In this hypothetical scenario, there are three isoforms transcribed (blue, green, red). Their transcripts are sequenced to produce reads (orange). Due to degradation, the observed reads are truncated, predominantly from the 5' end. This results in multi-mapping reads (starred). The number of possible isoforms each read is compatible with is indicated by the number of stars adjacent to the read.}
    \label{fig:degbias}
\end{figure}

A minor pathway for RNA decay involves endoribonuclease cleavage within the body of the transcript \cite{Garneau2007}. Endoribonuclease cleavage exposes unprotected 5’ and 3’ ends of the distal and proximal segments of the cleaved transcript, which then expose these segments to degradation by exoribonucleases \cite{ Gatfield2004,Garneau2007}. Differential rates of mRNA decay in different functional sets of genes have also been studied. For instance, transcription factor mRNAs tend to exhibit higher rates of mRNA decay compared to mRNA that code for proteins \cite{Yang2003}. Other classes of RNA also exhibit variable decay patterns \cite{Keene2010}; \cite{Kaiwan2021} described large variability in the decay rates of lncRNA compared to mRNAs, finding lncRNAs that were both extremely short-lived and long-lived. 

Compared to \textit{in vivo} effects, extra-cellular \textit{in vitro} effects are less well studied, due to the relative novelty of nanopore sequencing. In \cite{Workman2019}, it was suggested that read truncation in nanopore sequencing could be due to electronic signal noise from current spikes of unknown origin. In addition, pore blocking, which results in the stoppage of RNA translocation through the nanopore, could also result in truncated reads \cite{Amarasinghe2020}. Besides these considerations, generic factors that are not specific to nanopore sequencing, such as the fragmentation of transcripts during library preparation, can also result in the observed read truncation. 

Existing methods for transcript quantification from long-read RNA-seq, such as FLAIR \cite{Tang2020} and NanoCount \cite{Gleeson2021}, do not attempt to correct for degradation bias. To the best of our knowledge, the only method that currently models degradation bias is Bambu \cite{Bambu2022}, a tool for transcript discovery and quantification. To assign reads to isoforms, Bambu uses an alignment compatibility weight between the read class (reads sharing the same splice junctions) and isoform. In multi-mapping read classes, these weights are adjusted to account for constant degradation based on the effective length of the read class \cite{Bambu2022}. 

Keeping the ideas from this review in mind, we aim to characterise degradation in long-read direct RNA-seq and develop a model for correcting such bias.   

\section{Organisation}

The chapters of this thesis are organised as follows:
\begin{itemize}
    \item In Chapter 2, we formalise the notion of degradation, and characterise degradation in long-read direct RNA-seq data in reference isoforms and sequencing spike-ins across multiple cell lines and samples. 
    \item In Chapter 3, we develop a generative model for degradation-aware transcript quantification and derive an expectation maximization algorithm for parameter inference. 
    \item In Chapter 4, we evaluate our model and inference algorithms on simulated datasets with known degradation and real datasets with sequencing spike-ins. We benchmark our model against existing long-read transcript quantification methods in the literature. 
    \item In Chapter 5, we summarise the ideas of this thesis and discuss potential directions of future work. 
\end{itemize}